### DISCUSSION TITLE
Type errors using Annotated[some_type, ...] as type argument

### DISCUSSION THREAD

**Discussion by tdsmith** [hynek/svcs#74] [General]
Calls of the form

```python
with svcs.Registry() as reg:
    reg.register_value(Annotated[int, "my_int"], 42)
```

are recommended [by the svcs docs](https://svcs.hynek.me/en/stable/core-concepts.html#multiple-factories-for-the-same-type) for the case of multiple factories for the same type (here, `int`).

Pyright has recently started complaining about this, with a (slightly unhelpful) message of the form `error: Argument of type "type[int]" cannot be assigned to parameter "svc_type" of type "type" in function "register_value" (reportArgumentType)`. ([gist](https://gist.github.com/tdsmith/301461ba3b95c64bdea51bf9d7eea0c6))

Some interesting related conversations came up in a search:
https://github.com/microsoft/pyright/issues/7238 (Typing issue with Annotated)
https://discuss.python.org/t/is-annotated-compatible-with-type-t/43898 (Is `Annotated` compatible with `type[T]`?) (spoiler: "no")
https://github.com/python/typing-council/issues/18 (Typing spec change for Annotated special form)

I think this means that the svcs docs should not recommend the Annotated construct. I think the alternative is weakening the declared type of svc_type to Any, which doesn't seem great. Does that sound right to you?

**Comment by hynek** [comment]
Oh cool our Pyright config made it pointless for our test, because this is still passing:

https://github.com/hynek/svcs/blob/main/tests/typing/core.py#L144-L152

---

This is unfortunate but I guess the only way forward is to stop recommending Annotated? We might add a side-note I guess.

**Reply by hynek** [reply]
"fixed" in #92
